app/controllers/rubrics_controller.rb

```diff
@@ -61,8 +63,20 @@ def destroy
 
   def download_csv
     @assignment = Assignment.find(params[:assignment_id])
-    file_out = RubricCriterion.create_csv(@assignment)
-    send_data(file_out, type: 'text/csv', filename: "#{@assignment.short_identifier}_rubric_criteria.csv", disposition: 'inline')
+    file_out = MarkusCSV.generate(@assignment.rubric_criteria) do |criterion|
+      criterion_array = [criterion.rubric_criterion_name,criterion.weight]
+      (0..RUBRIC_LEVELS - 1).each do |i|
+        criterion_array.push(criterion['level_' + i.to_s + '_name'])
+      end
+      (0..RUBRIC_LEVELS - 1).each do |i|
+        criterion_array.push(criterion['level_' + i.to_s + '_description'])
+      end
+      criterion_array
+    end
+    send_data(file_out,
```

Comment:
This is more of a question for @david-yz-liu  
Should the send_data go in the MarkusCSV class?  It seems that we are unlikely to do anything with generated csv output other than send it.